### PR TITLE
For #385: Created tests for user activation

### PR DESCRIPTION
**Pull request by paulodamaso**
For #385: Created tests for user activation
- Created tests for accessing `/activate` with `GET` (error)
- Created tests for accessing `/activate` with `POST`, no user in session
- Created tests for accessing `/activate` with `POST`, no user in session
- Added puzzle

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#480](https://github.com/timelesslounge/timelessis/issues/480) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#480](https://github.com/timelesslounge/timelessis/issues/480) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/661176c2-e526-4df3-97bf-a46b29cb12db); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+     redirect to correct page. Fix this behavior and uncomment this test.
+    """
+    employee = factories.EmployeeFactory(
+        company=factories.CompanyFactory()
+    )
+    employee.account_status = False
+    g.user = employee
+    response = client.post("/auth/activate")
+    assert b"<h1>Successfully activated your account.</h1>" in response.data
+    assert employee.account_status == True
```
Comment (vryazanov) [inline tests/test_auth.py]: @paulodamaso 
python dev will never write like that :)
just `assert employee.account_status` (preferable) or  `assert employee.account_status is True`

File: tests/test_auth.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def test_activate_unauthenticated(client):
+    """ Tests if unauthenticated POST to activate returns correct screen"""
+    response = client.post("/auth/activate")
+    assert b"<h1>You are not logged in</h1>" in response.data
+    assert response.status_code ==  HTTPStatus.OK
+
+
+@pytest.mark.skip(reason="Test must set user in session")
```
Comment (vryazanov) [inline tests/test_auth.py]: @paulodamaso 
```python
    employee = factories.EmployeeFactory(
        company=factories.CompanyFactory(),
        ....
    )

    with client.session_transaction() as session:
        session["user_id"] = employee.id
```

Comment (paulodamaso) [reply tests/test_auth.py]: @vryazanov Didn't worked, unfortunately :(. It shows the following message:
```
AttributeError: 'SecureCookieSession' object has no attribute 'commit'
```

**Comment by paulodamaso** [comment]
@vryazanov Corrections done, and I have a comment, please take a look

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 15 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#480](https://github.com/timelesslounge/timelessis/issues/480) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vryazanov According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vryazanov** [comment]
@ypshenychka confirm

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**